pkg/supportbundle/execute.go

```diff
@@ -0,0 +1,164 @@
+// the executor closes the channel or the internal timeout expires.
+func executeUpdateRoutine(bundle *types.SupportBundle) chan interface{} {
+	progressChan := make(chan interface{})
+
+	timeout := time.After(60 * time.Minute)
+
+	logger.Infof("Executing Update go routine for support bundle ID: %s", bundle.ID)
+
+	go func() {
+		var collectorsComplete int
+		logger.Debugf("Waiting for %d collectors to complete", bundle.Progress.CollectorCount)
+
+		for {
+			select {
+			case msg, ok := <-progressChan:
+				if !ok {
+					logger.Debugf("Progress channel update closed for support bundle ID: %s", bundle.ID)
+
+					if !supportBundleIsComplete(bundle, collectorsComplete) {
+						logger.Errorf("Support bundle collection exited before completion for support bundle ID: %s", bundle.ID)
+						bundle.Status = "failed"
+
+						if err := store.GetStore().UpdateSupportBundle(bundle); err != nil {
+							logger.Error(errors.Wrap(err, "could not write failure to bundle"))
+							return
+						}
+					}
+					return
+				}
+
+				switch val := msg.(type) {
+				case error:
+					// Errors could be expected with RBAC, just log and continue
+					logger.Infof("Progress channel received an error, %s, for support bundle ID: %s", val, bundle.ID)
+
+				case troubleshootv1beta2.SupportBundleProgress:
+					if val.CurrentType == "collector" {
+						collectorsComplete++
+						logger.Debugf("Received collector progress update %d, %s, for support bundle ID: %s", collectorsComplete, val.CurrentName, bundle.ID)
+
+						bundle.Progress.CollectorsComplete = collectorsComplete
+						bundle.Progress.Message = val.CurrentName
+						if err := store.GetStore().UpdateSupportBundle(bundle); err != nil {
+							logger.Error(errors.Wrap(err, "could not update progress to bundle"))
+							return
+						}
+					} else {
+						logger.Errorf("Received unknown progress update for support bundle ID: %s", bundle.ID)
+					}
+
+				default:
+					logger.Errorf("Received unknown progress update, %v, of type %T, for support bundle ID: %s", val, val, bundle.ID)
+				}
+			case <-timeout:
+				logger.Errorf("Timeout exceeded for support bundle ID: %s", bundle.ID)
+				if err := store.GetStore().UpdateSupportBundle(bundle); err != nil {
+					logger.Error(errors.Wrap(err, "could not write failure to bundle"))
+					return
+				}
+				return
+			}
+		}
+	}()
+
+	return progressChan
+}
+
+// executeSupportBundleCollectRoutine creates a goroutine to collect the support bundle, upload, analyze and
+// send redactors. The function takes a channel for progress updates and closes it when collectors are complete.
+func executeSupportBundleCollectRoutine(bundle *types.SupportBundle, progressChan chan interface{}) {
+
+	collectorCB := func(c chan interface{}, msg string) {
+		c <- troubleshootv1beta2.SupportBundleProgress{
+			CurrentName:   msg,
+			CurrentStatus: "complete",
+			CurrentType:   "collector",
+		}
+	}
+
+	k8sconfig, err := k8sutil.GetRESTConfig()
```

Comment:
please use `k8sutil.GetClusterConfig()` instead

Reply:
I'll need to do some reading about how these are the same/different.